### DISCUSSION TITLE
Force revalidate `getStaticProps()`

### DISCUSSION THREAD

**Discussion by GMaiolo** [vercel/next.js#20829] [Help] [answered]
Hello, is it possible to force revalidating for `getStaticProps()` doing something like

```
return {
  revalidate: 0,
  props: {},
}
```

Or will that ignore revalidate as a whole?

**Comment by jamesmosier** [comment]
What do you mean by "force revalidating"? `revalidate: 0` will cause the page to be revalidated after 0 seconds so if someone visits you application they will be served the ["stale" version](https://nextjs.org/docs/basic-features/data-fetching#incremental-static-regeneration) and then after the page is regenerated they will be served that page.

**Comment by GMaiolo** [comment]
@jamesmosier
My use case is (kind of) like this

```ts
export const getStaticPaths = () => {
  return {
    paths: [],
    fallback: "blocking",
  }
}

export const getStaticProps = async () => {
  const foo = await getFoo()
  if (foo) {
    // when I have data I want normal behaviour
    return {
      revalidate: 7200,
      props: { foo },
    }
  }
  // otherwise I want it to force revalidate on next load
  return {
    revalidate: 0,
    props: {}
  }
}
```

**Comment by GMaiolo** [comment, marked as answer]
I just got this error

```
Error: A page's revalidate option can not be less than or equal to zero. A revalidate option of zero means to revalidate after _every_ request, and implies stale data cannot be tolerated.
To never revalidate, you can set revalidate to `false` (only ran once at build-time).
To revalidate as soon as possible, you can set the value to `1`.
```

So evidently  the minimum value is `1`.
